src/ipred_prepare.rs

```diff
@@ -131,7 +131,7 @@ pub unsafe fn rav1d_prepare_intra_edges<BD: BitDepth>(
     edge_flags: EdgeFlags,
     dst: *const BD::Pixel,
     stride: ptrdiff_t,
-    prefilter_toplevel_sb_edge: *const BD::Pixel,
+    prefilter_toplevel_sb_edge: &[BD::Pixel],
```

Comment:
This can be an `Option<&[BD::Pixel]>` since the only use of it is checked with `!prefilter_toplevel_sb_edge.is_null()`, so that can be an `if let Some(prefilter_toplevel_sb_edge) = prefilter_toplevel_sb_edge`.  Then you don't need to conflate empty slices with `None`.